### ISSUE TITLE
Seeded random samplers (determinism contract)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/geopatcher#18]
**Labels**: `enhancement`, `correctness`, `ml`, `epic:robustness`

### User story

As an ML researcher comparing two operator versions, I need bit-identical anchor sets between runs.

### Motivation

`SpatialRandom`, `SpatialJitteredStride`, `SpatialPoissonDisk`, `TemporalRandom` are all stochastic but have no explicit seed contract. Without it, runs are not reproducible — ML evaluation, regression tests, and journal-resume all silently break.

### Mathematics

Per-sampler RNG state. For sub-anchor independence, derive sub-keys deterministically:

- NumPy: `rng.spawn(n_anchors)` (NumPy ≥ 1.25) or `SeedSequence`.
- JAX: `jax.random.split(key, n_anchors)`.

### API proposal

```python
RNGLike = int | np.random.Generator | "jax.Array"

class SpatialJitteredStride:
    def __init__(self, step, jitter, rng: RNGLike = 0): ...

class SpatialRandom:
    def __init__(self, n_anchors, rng: RNGLike = 0): ...

class SpatialPoissonDisk:
    def __init__(self, radius, rng: RNGLike = 0): ...
```

`anchors_` is a cached property, deterministic given `rng`.

### Example usage

```python
sampler_a = gp.SpatialJitteredStride(step=(192, 192), jitter=(32, 32), rng=42)
sampler_b = gp.SpatialJitteredStride(step=(192, 192), jitter=(32, 32), rng=42)
assert sampler_a.anchors_ == sampler_b.anchors_  # bit-identical

# Resume after crash with same rng → same anchors → same outputs
```

### Implementation plan

- [ ] Add `rng` to every stochastic sampler constructor.
- [ ] Resolve `int → np.random.default_rng(int)` in `__post_init__`.
- [ ] Persist sampler state in `sampler.state_` for replay/debugging.
- [ ] `anchors_` becomes a `functools.cached_property`.
- [ ] Document the contract in `docs/patching/`: "Random samplers are deterministic given `rng`."

### Testing plan

- [ ] `sampler.anchors_ == sampler.anchors_` (idempotence).
- [ ] Two samplers with same `rng` produce identical anchor sets.
- [ ] Different `rng` produces statistically different sets (chi-square smoke test).
- [ ] Hypothesis: for any seed, anchor set is bit-identical across runs.
- [ ] JAX path: PRNGKey threaded correctly through `jax.random.split`.
